appcelerator-archive/amp
# Stacks 1.13 refactoring

Related to #733

#738 should be merged first

refactor amp stack to use Docker 1.13 way of managing them.

to test:

amp pf stop amp (stop amp old way)
make install (clean before could be required)
make TAG=local build-image

amp pf monitor (in the separated console)
amp pf start --local -> start only ampcore

// test amplog
amp logs -> error
amp pf start amplog --local (wait for stack started)
amp logs --infra -> a lot of logs...
amp pf stop amplog

// test ampmonitor
amp stats -> error
amp pf start ampmonitor --local (wait for stack started, and wait a while to have some data)
amp stats -> a lot of metrics
open a browser
type url: http://grafana.ampmonitor.local.appcelerator.io -> see grafana ui login (showing haproxy routes well the grafana ui using grafana label and ampmonitor stack)
amp pf stop ampmonitor


// test ampfunction
amp pf start ampfunction --local
cd examples/functions/demo-function
make image
amp fn create test appcelerator/amp-demo-function
cat Makefile | curl fn.ampfunction.local.appcelerator.io/test --data-binary @-
-> function executed (and haproxy route right fn label to function service)
amp pf stop ampfunction
cd ../../.. (back to amp home)

// test ampadmin
// add the line: adminserveraddress: 127.0.0.1:8081 in ~/.config/amp/amp.yml if it's not already done
amp pf start ampadmin --local (wait stack is started)
amp node ls -> see your host information 
amp node count -> see your stopped container
amp node purge --container -> container deleted
amp node count -> see you don't have stopped container anymore
amp pf stop ampadmin

// test user stack
amp stack up test -c api/rpc/stack/test_samples/sample-03.yml (wait stack is started)
curl www.test.local.appcelerator.io/ping -> pong (showing haproxy routes well to pinger service using 'www' label and 'test' stack))
amp stack ps test -> list task
amp stack services test -> list services
amp stack rm test
amp stack ps ampcore -> see it's working also for infra stack
amp stack services ampcore -> same

automated tests:

amp pf stop (wait all is down)
amp pf start all --local -v (wait all is up)
make test
